### DISCUSSION TITLE
How to change dynamically tailwind.config.js file color ?

### DISCUSSION THREAD

**Discussion by tasnimultas67** [tailwindlabs/tailwindcss#12647] [Help] [answered]
Hello there, I want when the user selects the color code, that color code will be dynamically replaced in the color section of the config file and the color of the full website will be changed. I can only get the color code in the console but can't set it dynamically in the config file. I am building a website with React.js

![Screenshot_2](https://github.com/tailwindlabs/tailwindcss/assets/72703360/e0567503-4ddf-4a02-9002-4635ee7d7a4c)
![Screenshot_3](https://github.com/tailwindlabs/tailwindcss/assets/72703360/70ef6289-20ee-4976-92d5-e6bb3d00ffbf)

**Comment by wongjn** [comment, marked as answer]
You could consider having the [color in the Tailwind configuration as a CSS variable](https://tailwindcss.com/docs/customizing-colors#using-css-variables) and then set the value of this CSS variable at runtime in the website.

**Reply by tasnimultas67** [reply]
Thank you sooo muchh ❤

**Comment by quedicesebas** [comment]
It is possible to set with variable the full palette, like this, but with css variables?:

```
primary: {
          '50': '#effaf4',
          '100': '#d7f4e3',
          '200': '#b3e7cc',
          '300': '#81d4ae',
          '400': '#4dba8b',
          '500': '#2a9f71',
          '600': '#1c7f5a',
          '700': '#16664a',
          '800': '#14513c',
          '900': '#114232',
          '950': '#09251d',
        },
```

**Reply by DeepakBalaraman** [reply]
Yes, all you have to do is set individual CSS variables for each color in the palette and define them in the tailwind config file.

### globals.css
```css
@layer base {
  :root {
    --color-primary-50: 240 247 255;
    --color-primary-100: 224 239 254;
    --color-primary-200: 185 223 254;
    --color-primary-300: 124 198 253;
    --color-primary-400: 54 170 250;
    --color-primary-500: 12 143 235;
    --color-primary-600: 0 120 215;
    --color-primary-700: 1 89 163;
    --color-primary-800: 6 76 134;
    --color-primary-900: 11 64 111;
    --color-primary-950: 7 40 74;
  }
}
```

### tailwind.config.js
```javascript
/* ... */
colors: {
        primary: {
          50: "rgb(var(--color-primary-50) / <alpha-value>)",
          100: "rgb(var(--color-primary-100) / <alpha-value>)",
          200: "rgb(var(--color-primary-200) / <alpha-value>)",
          300: "rgb(var(--color-primary-300) / <alpha-value>)",
          400: "rgb(var(--color-primary-400) / <alpha-value>)",
          500: "rgb(var(--color-primary-500) / <alpha-value>)",
          600: "rgb(var(--color-primary-600) / <alpha-value>)",
          700: "rgb(var(--color-primary-700) / <alpha-value>)",
          800: "rgb(var(--color-primary-800) / <alpha-value>)",
          900: "rgb(var(--color-primary-900) / <alpha-value>)",
          950: "rgb(var(--color-primary-950) / <alpha-value>)",
        },
},
/* ...  */
```
